react/Alert/Alert.less

```diff
@@ -42,6 +42,7 @@
   display: inline-block;
   vertical-align: middle;
   max-width: ~"calc(100% - (@{icon-width} + @{alert-gutter}))";
+  flex-grow: 1;
```

Comment:
Using `flex-grow` in this rule is a cleaner solution that the various max-width rules we have for `.text`. I think we can remove all the max width rules here?

Reply:
Just pushed an update—I was actually able to remove quite a few styles. The close icon moved a few pixels to the right, if that matters, but it looks okay to me.